CMD + click to open a file in terminal breaks if the file is above the terminal's cwd

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

If my terminal is at `root/dir/`, and I CMD + click a link to a file at `../file`, Zed will fail to open it, and create a non-existent reference to that file in the project explorer called `../file` that I cannot open. 

A scenarios that commonly causes this issue:
- in a git directory, make a change to a file at the root (ie, `root/file`)
![CleanShot 2024-05-01 at 21 58 53@2x](https://github.com/zed-industries/zed/assets/18580335/fff0afac-35e1-40ba-b9a8-fb927ba0e4dd)
- in the terminal, path to a directory at least one level deeper relative to the parent dir of the file that was changed (ie, `root/dir`)
- run `git status`, it prints that there's a change to `../file`
- cmd click on the path
- dummy file gets added to the explorer
![CleanShot 2024-05-01 at 22 00 37@2x](https://github.com/zed-industries/zed/assets/18580335/b9d95d4c-b18a-4f01-83b5-0b5a66a86279)
- the file cannot be opened
![CleanShot 2024-05-01 at 22 01 46@2x](https://github.com/zed-industries/zed/assets/18580335/cdbfb16c-4c34-48f0-ace0-b1a2d84cb3ee)


### Environment

```
Zed: v0.133.5 (Zed)
OS: macOS 14.4.1
Memory: 64 GiB
Architecture: aarch64
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

```
2024-05-01T21:43:23-04:00 [INFO] open paths ["/Users/-/Developer/repos/redacted/redacted/redacted/../tsconfig/tsconfig.json"]
2024-05-01T21:43:23-04:00 [WARN] Cannot absolutize "redacted/redacted/../tsconfig/tsconfig.json" received as Added FS change: invalid path
2024-05-01T21:43:28-04:00 [ERROR] invalid path
2024-05-01T22:01:36-04:00 [ERROR] invalid path
```

--- a/crates/terminal_view/src/terminal_view.rs
+++ b/crates/terminal_view/src/terminal_view.rs
@@ -787,23 +787,20 @@ fn possible_open_targets(
     let row = path_position.row;
     let column = path_position.column;
     let maybe_path = path_position.path;
-    let potential_abs_paths = if maybe_path.is_absolute() {
-        HashSet::from_iter([maybe_path])
+
+    let abs_path = if maybe_path.is_absolute() {
+        Some(maybe_path)
     } else if maybe_path.starts_with("~") {
-        if let Some(abs_path) = maybe_path
+        maybe_path
             .strip_prefix("~")
             .ok()
             .and_then(|maybe_path| Some(dirs::home_dir()?.join(maybe_path)))
-        {
-            HashSet::from_iter([abs_path])
-        } else {
-            HashSet::default()
-        }
     } else {
-        // First check cwd and then workspace
         let mut potential_cwd_and_workspace_paths = HashSet::default();
         if let Some(cwd) = cwd {
-            potential_cwd_and_workspace_paths.insert(Path::join(cwd, &maybe_path));
+            let abs_path = Path::join(cwd, &maybe_path);
+            let canonicalized_path = abs_path.canonicalize().unwrap_or(abs_path);
+            potential_cwd_and_workspace_paths.insert(canonicalized_path);
         }
         if let Some(workspace) = workspace.upgrade() {
             workspace.update(cx, |workspace, cx| {
@@ -815,10 +812,25 @@ fn possible_open_targets(
                 }
             });
         }
-        potential_cwd_and_workspace_paths
+
+        return possible_open_paths_metadata(
+            fs,
+            row,
+            column,
+            potential_cwd_and_workspace_paths,
+            cx,
+        );
     };
 
-    possible_open_paths_metadata(fs, row, column, potential_abs_paths, cx)
+    let canonicalized_paths = match abs_path {
+        Some(abs_path) => match abs_path.canonicalize() {
+            Ok(path) => HashSet::from_iter([path]),
+            Err(_) => HashSet::default(),
+        },
+        None => HashSet::default(),
+    };
+
+    possible_open_paths_metadata(fs, row, column, canonicalized_paths, cx)
 }
 
 fn regex_to_literal(regex: &str) -> String {

